lib/datasource/deb/index.ts

```diff
@@ -0,0 +1,248 @@
+   * the Debian package repository format as specified here
+   * @see{https://wiki.debian.org/DebianRepository/Format}
+   */
+  override readonly customRegistrySupport = true;
+
+  /**
+   * deb uri distribution [component1] [component2] [...]
+   * @see{https://wiki.debian.org/DebianRepository/Format}
+   */
+  override readonly defaultRegistryUrls = [
+    'deb https://ftp.debian.org/debian stable main contrib non-free',
+  ];
+
+  override readonly defaultConfig: DebLanguageConfig = {
+    deb: {
+      binaryArch: 'amd64',
+      downloadDirectory: '/tmp/renovate-deb/packages-download',
+      extractionDirectory: '/tmp/renovate-deb/packages',
+    },
+  };
+
+  override readonly caching = false; // TODO: how can this be used?
+
+  /**
+   * Here, we tell Renovate that this data source can respect multiple upstream repositories
+   */
+  override readonly registryStrategy = 'merge';
+
+  requiredPackageKeys = ['Package', 'Version', 'Homepage'];
+
+  initCacheDir(cfg: DebLanguageConfig): void {
+    mkdirSync(cfg.deb.downloadDirectory, { recursive: true });
+    mkdirSync(cfg.deb.extractionDirectory, { recursive: true });
+  }
+
+  async runCommand(wd: string, command: string): Promise<void> {
+    const pexec = promisify(exec);
+    logger.trace('running command ' + command + ' in directory ' + wd);
+    const { stdout, stderr } = await pexec(
+      ['cd', '"' + wd + '"', '&&', command].join(' ')
+    );
+    logger.debug(stdout);
+    logger.debug(stderr);
+  }
+
+  async downloadAndExtractPackage(
+    cfg: DebLanguageConfig,
+    packageUrl: string
+  ): Promise<string> {
+    // we hash the package URL and export the hex to make it file system friendly
+    // we use the hashed url as the filename for the local directories/files
+    const hash = createHash('sha256');
+    hash.update(packageUrl);
+    const hashedPackageUrl = hash.copy().digest('hex');
+    const downloadLocation = cfg.deb.downloadDirectory + '/' + hashedPackageUrl;
+    const compressedFile = downloadLocation + '/Packages.xz';
+    const extractedFile =
+      cfg.deb.extractionDirectory + '/' + hashedPackageUrl + '.txt';
+
+    // curl will not modify the local file, if the upstream HTTP server
+    // does not return a modified version. We can use this information
+    // to not re-extract the same file again. For this to work we need
+    // the current modification timestamp of the local package file.
+    let lastTimestamp = -1;
+    try {
+      const stats = statSync(compressedFile);
+      lastTimestamp = stats.mtime.getTime();
+    } catch (e) {
+      // ignore if the file doesnt exist
+    }
+
+    logger.debug(
+      'Downloading package file from ' + packageUrl + ' as ' + downloadLocation
+    );
+    /**
+     * curl -o "$file" -z "$file" "$uri"
+     */
+    mkdirSync(downloadLocation, { recursive: true });
+    await this.runCommand(
+      downloadLocation,
+      ['curl', '-o', 'Packages.xz', '-z', 'Packages.xz', packageUrl].join(' ')
```

Comment:
nope, use `this.http.stream` to download files, see terraform manager (check the lockfile dir)